# Akron, Ohio Library Location

### By: Helena Donnelly

### First, I used Foursquare API to get the locations of all libraries in and around Akron, Ohio.

In [1]:
# The code was removed by Watson Studio for sharing.

In [2]:
import requests
url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&near={}&limit={}&categoryId={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            'Akron, OH',  
            100,
            #The category ID is associated with libraries
            '4bf58dd8d48988d12f941735')
results = requests.get(url).json()["response"]['groups'][0]['items']
results

[{'reasons': {'count': 0,
   'items': [{'summary': 'This spot is popular',
     'type': 'general',
     'reasonName': 'globalInteractionReason'}]},
  'venue': {'id': '4b43e4d9f964a52068ed25e3',
   'name': 'Akron-Summit County Public Library: Main Branch',
   'location': {'address': '60 S High St',
    'crossStreet': 'at E Market',
    'lat': 41.08336493528707,
    'lng': -81.51683941671816,
    'labeledLatLngs': [{'label': 'display',
      'lat': 41.08336493528707,
      'lng': -81.51683941671816}],
    'postalCode': '44326',
    'cc': 'US',
    'city': 'Akron',
    'state': 'OH',
    'country': 'United States',
    'formattedAddress': ['60 S High St (at E Market)',
     'Akron, OH 44326',
     'United States']},
   'categories': [{'id': '4bf58dd8d48988d12f941735',
     'name': 'Library',
     'pluralName': 'Libraries',
     'shortName': 'Library',
     'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/building/library_',
      'suffix': '.png'},
     'primary': True}],
   'ph

### I then made the data I received into a pandas dataframe.

In [3]:
from pandas.io.json import json_normalize
libraries = json_normalize(results)
libraries

flags.outsideRadius  reasons.count  \
0                  NaN              0   
1                 True              0   
2                  NaN              0   
3                  NaN              0   
4                 True              0   
5                  NaN              0   
6                 True              0   
7                 True              0   
8                 True              0   
9                  NaN              0   
10                True              0   
11                True              0   
12                True              0   
13                True              0   
14                True              0   
15                 NaN              0   
16                True              0   
17                 NaN              0   
18                 NaN              0   
19                 NaN              0   
20                 NaN              0   
21                 NaN              0   
22                 NaN              0   
23                 NaN              0   
24                 NaN              0   
25                True              0   
26                True              0   
27                True              0   
28                True              0   
29                True              0   
30                True              0   
31                True              0   
32                True              0   
33                 NaN              0   
34                True              0   
35                True              0   
36                True              0   
37                 NaN              0   
38                 NaN              0   
39                 NaN              0   
40                 NaN              0   
41                True              0   
42                True              0   
43                True              0   
44                True              0   
45                True              0   
46                 NaN              0   
47                True              0   
48                 NaN              0   

                                        reasons.items  \
0   [{'summary': 'This spot is popular', 'type': '...   
1   [{'summary': 'This spot is popular', 'type': '...   
2   [{'summary': 'This spot is popular', 'type': '...   
3   [{'summary': 'This spot is popular', 'type': '...   
4   [{'summary': 'This spot is popular', 'type': '...   
5   [{'summary': 'This spot is popular', 'type': '...   
6   [{'summary': 'This spot is popular', 'type': '...   
7   [{'summary': 'This spot is popular', 'type': '...   
8   [{'summary': 'This spot is popular', 'type': '...   
9   [{'summary': 'This spot is popular', 'type': '...   
10  [{'summary': 'This spot is popular', 'type': '...   
11  [{'summary': 'This spot is popular', 'type': '...   
12  [{'summary': 'This spot is popular', 'type': '...   
13  [{'summary': 'This spot is popular', 'type': '...   
14  [{'summary': 'This spot is popular', 'type': '...   
15  [{'summary': 'This spot is popular', 'type': '...   
16  [{'summary': 'This spot is popular', 'type': '...   
17  [{'summary': 'This spot is popular', 'type': '...   
18  [{'summary': 'This spot is popular', 'type': '...   
19  [{'summary': 'This spot is popular', 'type': '...   
20  [{'summary': 'This spot is popular', 'type': '...   
21  [{'summary': 'This spot is popular', 'type': '...   
22  [{'summary': 'This spot is popular', 'type': '...   
23  [{'summary': 'This spot is popular', 'type': '...   
24  [{'summary': 'This spot is popular', 'type': '...   
25  [{'summary': 'This spot is popular', 'type': '...   
26  [{'summary': 'This spot is popular', 'type': '...   
27  [{'summary': 'This spot is popular', 'type': '...   
28  [{'summary': 'This spot is popular', 'type': '...   
29  [{'summary': 'This spot is popular', 'type': '...   
30  [{'summary': 'This spot is popular', 'type': '...   
31  [{'summary': 'This spot is popular', 'type': '...   
32  [{'summary': 'This spot is popular', 'type': '...   
33  [{'summary

### I then selected only the latitude, longitude and venue.name columns and removed rows which were not public libraries before reseting the index.

In [4]:
import pandas as pd
libraries_info_prelim = libraries[['venue.location.lat','venue.location.lng','venue.name']]

In [5]:
not_public = ['Auburn Science Library','William McKinley Presidential Library & Museum','Conrad Ott Staff Development Center','Library Shop on Main','Akron Bookmobile','Bierce Library','The University of Akron Law Library', 'Science And Engineering Library']
libraries_mp = libraries_info_prelim[~libraries_info_prelim['venue.name'].isin(not_public)]
libraries_mp.reset_index(inplace = True)
libraries_mp

index  venue.location.lat  venue.location.lng  \
0       0           41.083365          -81.516839   
1       1           41.134771          -81.485502   
2       2           41.115915          -81.574477   
3       3           41.054888          -81.440852   
4       4           41.160995          -81.440889   
5       5           41.136310          -81.621572   
6       6           41.241590          -81.443457   
7       7           41.314069          -81.626122   
8       8           41.331228          -81.721279   
9       9           41.042591          -81.514704   
10     10           41.239848          -81.817134   
11     11           41.137700          -81.861865   
12     12           40.807501          -81.457350   
13     13           41.383238          -81.732032   
14     14           41.404598          -81.697316   
15     15           41.043814          -81.558175   
16     16           41.376848          -81.646840   
17     18           41.070995          -81.544334   
18     19           41.070995          -81.544334   
19     21           41.108272          -81.509716   
20     22           41.096652          -81.543462   
21     23           41.036160          -81.564450   
22     24           41.066634          -81.481187   
23     25           41.434109          -81.735940   
24     26           41.495701          -81.564678   
25     27           41.440737          -81.526012   
26     28           41.501488          -81.690829   
27     29           41.465750          -81.566419   
28     30           41.484666          -81.804208   
29     31           41.555777          -81.440938   
30     32           41.413617          -81.925711   
31     34           41.447664          -81.924587   
32     35           40.797588          -81.942049   
33     36           41.443797          -81.855891   
34     37           41.083678          -81.565000   
35     41           41.665376          -81.346765   
36     42           41.031620          -81.639250   
37     43           41.509895          -82.012773   
38     44           41.639450          -81.407296   
39     45           41.727680          -81.243029   
40     48           41.109375          -81.568264   

                                           venue.name  
0     Akron-Summit County Public Library: Main Branch  
1                              Cuyahoga Falls Library  
2              Northwest Akron Branch Library (ASCPL)  
3          Akron Summit County Library - Ellet Branch  
4                    Stow Munroe Falls Public Library  
5   Akron-Summit County Public Library - Fairlawn/...  
6                 Hudson Library & Historical Society  
7   Cuyahoga County Public Library - Brecksville B...  
8   Cuyahoga County Public Library - North Royalto...  
9                              Firestone Park Library  
10  Medina County District Library - Brunswick Branch  
11     Medina County District Library - Medina Branch  
12  Perry Sippo Branch - Stark County District Lib...  
13      Cuyahoga County Public Library - Parma Branch  
14  Cuyahoga County Public Library - Parma-Snow Br...  
15        Akron-Summit Public Library: Kenmore Branch  
16  Cuyahoga County Public Library - Independence ...  
17  Akron-Summit County Public Library: Odom Boule...  
18  Akron-Summit County Public Library: Odom Boule...  
19  Akron-Summit County Public Library–North Hill ...  
20  Akron-Summit County Public Library: Highland S...  
21         Akron summit county library Kenmore branch  
22  Akron-Summit County Public Library: Goodyear B...  
23   Cuyahoga County Public Library - Brooklyn Branch  
24  Cleveland Heights-University Heights Public Li...  
25  Cuyahoga County Public Library-Warrensville He...  
26             Cleveland Public Library - Main Branch  
27        Shaker Heights Public Library - Main Branch  
28              Lakewood Public Library - Main Branch  
29                           Mayfield Village Library  
30  Cuyahoga County Public Library - Nort

In [6]:
libraries_ul = libraries_mp[['venue.location.lat','venue.location.lng','venue.name']]
libraries_ul

venue.location.lat  venue.location.lng  \
0            41.083365          -81.516839   
1            41.134771          -81.485502   
2            41.115915          -81.574477   
3            41.054888          -81.440852   
4            41.160995          -81.440889   
5            41.136310          -81.621572   
6            41.241590          -81.443457   
7            41.314069          -81.626122   
8            41.331228          -81.721279   
9            41.042591          -81.514704   
10           41.239848          -81.817134   
11           41.137700          -81.861865   
12           40.807501          -81.457350   
13           41.383238          -81.732032   
14           41.404598          -81.697316   
15           41.043814          -81.558175   
16           41.376848          -81.646840   
17           41.070995          -81.544334   
18           41.070995          -81.544334   
19           41.108272          -81.509716   
20           41.096652          -81.543462   
21           41.036160          -81.564450   
22           41.066634          -81.481187   
23           41.434109          -81.735940   
24           41.495701          -81.564678   
25           41.440737          -81.526012   
26           41.501488          -81.690829   
27           41.465750          -81.566419   
28           41.484666          -81.804208   
29           41.555777          -81.440938   
30           41.413617          -81.925711   
31           41.447664          -81.924587   
32           40.797588          -81.942049   
33           41.443797          -81.855891   
34           41.083678          -81.565000   
35           41.665376          -81.346765   
36           41.031620          -81.639250   
37           41.509895          -82.012773   
38           41.639450          -81.407296   
39           41.727680          -81.243029   
40           41.109375          -81.568264   

                                           venue.name  
0     Akron-Summit County Public Library: Main Branch  
1                              Cuyahoga Falls Library  
2              Northwest Akron Branch Library (ASCPL)  
3          Akron Summit County Library - Ellet Branch  
4                    Stow Munroe Falls Public Library  
5   Akron-Summit County Public Library - Fairlawn/...  
6                 Hudson Library & Historical Society  
7   Cuyahoga County Public Library - Brecksville B...  
8   Cuyahoga County Public Library - North Royalto...  
9                              Firestone Park Library  
10  Medina County District Library - Brunswick Branch  
11     Medina County District Library - Medina Branch  
12  Perry Sippo Branch - Stark County District Lib...  
13      Cuyahoga County Public Library - Parma Branch  
14  Cuyahoga County Public Library - Parma-Snow Br...  
15        Akron-Summit Public Library: Kenmore Branch  
16  Cuyahoga County Public Library - Independence ...  
17  Akron-Summit County Public Library: Odom Boule...  
18  Akron-Summit County Public Library: Odom Boule...  
19  Akron-Summit County Public Library–North Hill ...  
20  Akron-Summit County Public Library: Highland S...  
21         Akron summit county library Kenmore branch  
22  Akron-Summit County Public Library: Goodyear B...  
23   Cuyahoga County Public Library - Brooklyn Branch  
24  Cleveland Heights-University Heights Public Li...  
25  Cuyahoga County Public Library-Warrensville He...  
26             Cleveland Public Library - Main Branch  
27        Shaker Heights Public Library - Main Branch  
28              Lakewood Public Library - Main Branch  
29                           Mayfield Village Library  
30  Cuyahoga County Public Library - North Olmsted...  
31                     Westlake Porter Public Library  
32                        Wayne County Public Library  
33  Cuyahoga County Public Library - Fairview Park...  
34           Akron-Summit County Library-Maple Valley  
35                Mentor Public Library - Main Branch  

### I then renamed the columns in the dataframe to be more simple

In [7]:
library_info = libraries_ul.rename(columns={"venue.location.lat": "latitude", "venue.location.lng": "longitude", "venue.name": "name"})
library_info


latitude  longitude                                               name
0   41.083365 -81.516839    Akron-Summit County Public Library: Main Branch
1   41.134771 -81.485502                             Cuyahoga Falls Library
2   41.115915 -81.574477             Northwest Akron Branch Library (ASCPL)
3   41.054888 -81.440852         Akron Summit County Library - Ellet Branch
4   41.160995 -81.440889                   Stow Munroe Falls Public Library
5   41.136310 -81.621572  Akron-Summit County Public Library - Fairlawn/...
6   41.241590 -81.443457                Hudson Library & Historical Society
7   41.314069 -81.626122  Cuyahoga County Public Library - Brecksville B...
8   41.331228 -81.721279  Cuyahoga County Public Library - North Royalto...
9   41.042591 -81.514704                             Firestone Park Library
10  41.239848 -81.817134  Medina County District Library - Brunswick Branch
11  41.137700 -81.861865     Medina County District Library - Medina Branch
12  40.807501 -81.457350  Perry Sippo Branch - Stark County District Lib...
13  41.383238 -81.732032      Cuyahoga County Public Library - Parma Branch
14  41.404598 -81.697316  Cuyahoga County Public Library - Parma-Snow Br...
15  41.043814 -81.558175        Akron-Summit Public Library: Kenmore Branch
16  41.376848 -81.646840  Cuyahoga County Public Library - Independence ...
17  41.070995 -81.544334  Akron-Summit County Public Library: Odom Boule...
18  41.070995 -81.544334  Akron-Summit County Public Library: Odom Boule...
19  41.108272 -81.509716  Akron-Summit County Public Library–North Hill ...
20  41.096652 -81.543462  Akron-Summit County Public Library: Highland S...
21  41.036160 -81.564450         Akron summit county library Kenmore branch
22  41.066634 -81.481187  Akron-Summit County Public Library: Goodyear B...
23  41.434109 -81.735940   Cuyahoga County Public Library - Brooklyn Branch
24  41.495701 -81.564678  Cleveland Heights-University Heights Public Li...
25  41.440737 -81.526012  Cuyahoga County Public Library-Warrensville He...
26  41.501488 -81.690829             Cleveland Public Library - Main Branch
27  41.465750 -81.566419        Shaker Heights Public Library - Main Branch
28  41.484666 -81.804208              Lakewood Public Library - Main Branch
29  41.555777 -81.440938                           Mayfield Village Library
30  41.413617 -81.925711  Cuyahoga County Public Library - North Olmsted...
31  41.447664 -81.924587                     Westlake Porter Public Library
32  40.797588 -81.942049                        Wayne County Public Library
33  41.443797 -81.855891  Cuyahoga County Public Library - Fairview Park...
34  41.083678 -81.565000           Akron-Summit County Library-Maple Valley
35  41.665376 -81.346765                Mentor Public Library - Main Branch
36  41.031620 -81.639250                              Norton Branch Library
37  41.509895 -82.012773                           Avon Lake Public Library
38  41.639450 -81.407296                          Willoughby Public Library
39  41.727680 -81.243029                                     Morley Library
40  41.109375 -81.568264                                   NW Akron Library

### I then imported folium and made an empty map focused on Akron, Ohio.

In [8]:
! pip install folium

     |████████████████████████████████| 102kB 5.5MB/s ta 0:00:01


In [9]:
import folium

libraries_map = folium.Map(location = [41.0814447,-81.5190053], zoom_start = 11)
libraries_map


### I then added markers to the map at the location of each of the libraries in and near Akron.

In [10]:
latitude = list(library_info['latitude'])
longitude = list(library_info['longitude'])
name = list(library_info['name'])
number = 0
while number < 41:
    folium.Marker(
        location=[latitude[number], longitude[number]],
        popup= name[number],
        tooltip = "Click for Library Name"
    ).add_to(libraries_map)
    number += 1
libraries_map

### I then installed geopandas and used a geojson file of the borders of Akron before adding them to the map

In [11]:
! pip install geopandas

     |████████████████████████████████| 972kB 6.3MB/s eta 0:00:01
     |████████████████████████████████| 14.7MB 12.0MB/s eta 0:00:01
     |████████████████████████████████| 1.8MB 18.1MB/s eta 0:00:01
     |████████████████████████████████| 10.9MB 13.0MB/s eta 0:00:01


In [12]:
import urllib.request
#https://nominatim.openstreetmap.org/ui/search.html?q=Akron%2C+Ohio
x = urllib.request.urlopen('http://polygons.openstreetmap.fr/get_geojson.py?id=182644&params=0')
akron_geo_prelim = x.read()


In [13]:

libraries_map.choropleth(geo_data= r'http://polygons.openstreetmap.fr/get_geojson.py?id=182644&params=0', fill_opacity = 0)


In [14]:
libraries_map

### Next, I got the numbers of each census tract in Akron (note - census tracts do not line up with city borders but at least part of each is in Akron) and then I used another census file to determine the latitude and longitude of each census tract  in Ohio and its FIPS code.

In [15]:
#From data.census.gov
census_tracts_in_Akron = [5011.00,  5012.00,  5013.01,  5013.02,  5017.00,  5018.00,  5019.00,  5021.01,  5021.02,  5022.00,  5023.00,  5024.00,  5025.00,  5026.00,  5027.00,  5028.00 ,  5031.00,  5032.00,  5033.00,  5034.00,  5035.00,  5036.00,  5037.01 ,  5037.02 ,  5038.00 ,  5041.00,  5042.00,  5044.00,  5045.00,  5046.00,  5047.00,  5048.00,  5051.00,  5052.00,  5053.00,  5054.00,  5055.00,  5056.00,  5057.00,  5058.00,  5059.00,  5061.00 ,  5062.00 ,  5063.02,  5063.03,  5063.04,  5064.00,  5065.00,  5066.00,  5067.00,  5068.00,  5069.00,  5071.01,  5071.02 ,  5072.01,  5072.02,  5072.03 ,  5073.00,  5074.00,  5075.00 ,  5076.00,  5080.00 ,  5083.01,  5083.99,  5086.00,  5088.00,  5089.00,  5090.00,  5203.00 ,  5203.01 ,  5310.01 ,  5310.02 ,  5311.01 ,  5318.01 ,  5318.02 ,  5322.02 ,  5323.02 ,  5329.00 ,  5329.01 ,  5329.02 ,  5330.00 ,  5334.00]

In [16]:
url = "https://www2.census.gov/geo/docs/maps-data/data/gazetteer/2019_Gazetteer/2019_gaz_tracts_39.txt"
latitude = []
longitude = []
Fips = []
file = urllib. request. urlopen(url)
for line in file:
    decoded_line = line. decode("utf-8")
    list_of_values = decoded_line.split()
    latitude.append(list_of_values[6])
    longitude.append(list_of_values[7])
    Fips.append(list_of_values[1])
final_latitude = latitude[1:]
final_longitude = longitude[1:]
final_Fips = Fips[1:]

In [17]:
census_dataframe = pd.DataFrame(list(zip(final_latitude, final_longitude, final_Fips)), columns = ['latitude', 'longitude','fips number'])
census_dataframe

latitude    longitude  fips number
0     38.9560042  -83.3557407  39001770100
1     38.9909749  -83.5518337  39001770200
2     38.8408353  -83.5833171  39001770300
3     38.7716214  -83.5445111  39001770400
4     38.7507835  -83.3646628  39001770500
5     38.7060421  -83.5610405  39001770600
6     40.8917469  -83.8991177  39003010100
7     40.8652862   -84.003472  39003010200
8     40.8240378  -84.1599836  39003010300
9     40.7249174  -84.3146099  39003010600
10    40.7732053  -84.2056584  39003010800
11    40.7446294  -84.1581047  39003010900
12    40.7773655  -84.1355805  39003011000
13    40.7823566  -84.0783177  39003011200
14    40.7728631  -84.0340979  39003011300
15    40.7810576  -83.9205969  39003011400
16     40.695518  -83.9280532  39003011500
17    40.6751135  -84.0414253  39003011600
18     40.718787   -84.145148  39003011800
19    40.6841495  -84.1343291  39003011900
20    40.7070395  -84.1683471  39003012000
21    40.6822427  -84.1973201  39003012100
22    40.7647074  -84.1169163  39003012200
23    40.7531926  -84.1170506  39003012300
24    40.7582635  -84.1040908  39003012400
25    40.7390385  -84.0838425  39003012600
26    40.7336078  -84.0945137  39003012700
27    40.7440996  -84.1194763  39003012900
28    40.7491919  -84.1356573  39003013000
29     40.738817  -84.1421507  39003013100
...          ...          ...          ...
2922  41.5381047  -83.6032897  39173020402
2923  41.5596694  -83.6302308  39173020500
2924  41.5351218  -83.6279124  39173020601
2925  41.5364228  -83.6664874  39173020602
2926  41.5257931  -83.5698204  39173020700
2927  41.5847185  -83.4975553  39173020800
2928  41.5516314  -83.4680267  39173020900
2929  41.4772021  -83.4602741  39173021000
2930  41.4056518  -83.5017788  39173021100
2931  41.4662766  -83.6677403  39173021200
2932   41.408529  -83.7917402  39173021300
2933  41.3217495   -83.820206  39173021400
2934  41.3202542  -83.7182594  39173021500
2935  41.4047869  -83.6729864  39173021600
2936  41.3802175  -83.6493367  39173021701
2937  41.3697478  -83.6470496  39173021702
2938   41.379731  -83.6295647  39173021800
2939   41.361645  -83.6775591  39173021901
2940  41.3597575  -83.6292108  39173021902
2941  41.2989166  -83.4769125  39173022100
2942  41.2135794  -83.4877868  39173022200
2943  41.2155448  -83.7188454  39173022300
2944  41.1821984  -83.6790476  39173022400
2945  41.3335215  -83.6012943  39173022500
2946  40.9140422  -83.2308074  39175938000
2947  40.9608887  -83.3943088  39175938100
2948  40.8103048  -83.4097539  39175938200
2949   40.850862  -83.2876057  39175938300
2950  40.8056711  -83.2811002  39175938400
2951   40.761518  -83.2142529  39175938500

[2952 rows x 3 columns]

### Since the first five numbers in a FIPS code refer to the state and county, I filtered out all rows which did not start with the numbers 39153, meaning they were not in Summit County, Ohio.

In [18]:
census_tracts_Summit_county = census_dataframe.loc[census_dataframe['fips number'].str.contains('39153')]
census_tracts_Summit_county.reset_index(inplace = True)
census_tracts = census_tracts_Summit_county.drop(['index'], axis=1)
census_tracts

latitude    longitude  fips number
0    41.0903877  -81.5058552  39153501100
1    41.0683703  -81.5140053  39153501700
2    41.0688191  -81.5429231  39153501800
3    41.0656521  -81.5326908  39153501900
4    41.1091563   -81.499983  39153502101
5    41.1097084   -81.478352  39153502102
6    41.0965811  -81.4897971  39153502200
7    41.0829912  -81.4760965  39153502300
8    41.0697756  -81.4811058  39153502500
9    41.0687054   -81.469198  39153502600
10   41.0788732  -81.4500049  39153502700
11    41.068563  -81.4470118  39153502800
12    41.068852  -81.4967052  39153503100
13   41.0601438  -81.4955638  39153503200
14    41.040496  -81.4971586  39153503300
15    41.063034   -81.486175  39153503400
16    41.049353  -81.4838909  39153503500
17   41.0533235  -81.4659917  39153503600
18   41.0522139  -81.4461962  39153503701
19   41.0526704  -81.4303702  39153503702
20   41.0291706  -81.4727794  39153503800
21    41.065477  -81.5056081  39153504100
22   41.0606598  -81.5139659  39153504200
23   41.0499564  -81.5233459  39153504400
24   41.0510032  -81.5142789  39153504500
25   41.0536023  -81.5050114  39153504600
26   41.0432782  -81.5135793  39153504700
27   41.0343632   -81.516324  39153504800
28   41.0615779  -81.5458687  39153505200
29    41.058939  -81.5354627  39153505300
..          ...          ...          ...
105  40.9457792  -81.5817439  39153531702
106  41.0184964   -81.537607  39153531801
107  41.0094637  -81.5074864  39153531802
108  41.0086144   -81.649467  39153532001
109  41.0451103  -81.6648842  39153532003
110  41.0440182  -81.6127542  39153532004
111  41.1243788  -81.6176923  39153532202
112   41.177394  -81.6627095  39153532301
113  41.1616527  -81.6109586  39153532302
114  41.2577903  -81.6401234  39153532501
115  41.2228382   -81.631974  39153532502
116  41.2417357   -81.540333  39153532600
117  41.3008659  -81.4858535  39153532701
118  41.3388568  -81.5268406  39153532702
119  41.2874758  -81.5325423  39153532703
120  41.3340145  -81.5655108  39153532705
121  41.3008741  -81.5650318  39153532706
122  41.3269173  -81.4922085  39153532708
123  41.1828813  -81.5530077  39153532901
124  41.1771447  -81.5109597  39153532902
125  40.9547135  -81.4360888  39153532999
126  41.0164309  -81.5664915  39153533000
127  41.2657958    -81.42083  39153533101
128   41.244439  -81.4167448  39153533102
129  41.2183195   -81.416403  39153533200
130  41.0808405  -81.6157451  39153533400
131  41.1250983  -81.6621756  39153533501
132  41.0904593  -81.6631372  39153533502
133  41.2268768  -81.4707843  39153534000
134  41.2662129   -81.464497  39153534100

[135 rows x 3 columns]

### Since the last six digits of the FIPS code is the census tract number times 100, I removed the first five digits from each FIPS code and divided it by 100 so it would match the census tract numbers I already had.

In [19]:
def shorten_fips(row):
    return(row[5:])
census_tracts['fips number']=census_tracts['fips number'].apply(shorten_fips)
census_tracts

latitude    longitude fips number
0    41.0903877  -81.5058552      501100
1    41.0683703  -81.5140053      501700
2    41.0688191  -81.5429231      501800
3    41.0656521  -81.5326908      501900
4    41.1091563   -81.499983      502101
5    41.1097084   -81.478352      502102
6    41.0965811  -81.4897971      502200
7    41.0829912  -81.4760965      502300
8    41.0697756  -81.4811058      502500
9    41.0687054   -81.469198      502600
10   41.0788732  -81.4500049      502700
11    41.068563  -81.4470118      502800
12    41.068852  -81.4967052      503100
13   41.0601438  -81.4955638      503200
14    41.040496  -81.4971586      503300
15    41.063034   -81.486175      503400
16    41.049353  -81.4838909      503500
17   41.0533235  -81.4659917      503600
18   41.0522139  -81.4461962      503701
19   41.0526704  -81.4303702      503702
20   41.0291706  -81.4727794      503800
21    41.065477  -81.5056081      504100
22   41.0606598  -81.5139659      504200
23   41.0499564  -81.5233459      504400
24   41.0510032  -81.5142789      504500
25   41.0536023  -81.5050114      504600
26   41.0432782  -81.5135793      504700
27   41.0343632   -81.516324      504800
28   41.0615779  -81.5458687      505200
29    41.058939  -81.5354627      505300
..          ...          ...         ...
105  40.9457792  -81.5817439      531702
106  41.0184964   -81.537607      531801
107  41.0094637  -81.5074864      531802
108  41.0086144   -81.649467      532001
109  41.0451103  -81.6648842      532003
110  41.0440182  -81.6127542      532004
111  41.1243788  -81.6176923      532202
112   41.177394  -81.6627095      532301
113  41.1616527  -81.6109586      532302
114  41.2577903  -81.6401234      532501
115  41.2228382   -81.631974      532502
116  41.2417357   -81.540333      532600
117  41.3008659  -81.4858535      532701
118  41.3388568  -81.5268406      532702
119  41.2874758  -81.5325423      532703
120  41.3340145  -81.5655108      532705
121  41.3008741  -81.5650318      532706
122  41.3269173  -81.4922085      532708
123  41.1828813  -81.5530077      532901
124  41.1771447  -81.5109597      532902
125  40.9547135  -81.4360888      532999
126  41.0164309  -81.5664915      533000
127  41.2657958    -81.42083      533101
128   41.244439  -81.4167448      533102
129  41.2183195   -81.416403      533200
130  41.0808405  -81.6157451      533400
131  41.1250983  -81.6621756      533501
132  41.0904593  -81.6631372      533502
133  41.2268768  -81.4707843      534000
134  41.2662129   -81.464497      534100

[135 rows x 3 columns]

In [20]:
def divide_by_100(row):
    row = float(row)
    return(row/100)
census_tracts['fips number']=census_tracts['fips number'].apply(divide_by_100)
census_tracts

latitude    longitude  fips number
0    41.0903877  -81.5058552      5011.00
1    41.0683703  -81.5140053      5017.00
2    41.0688191  -81.5429231      5018.00
3    41.0656521  -81.5326908      5019.00
4    41.1091563   -81.499983      5021.01
5    41.1097084   -81.478352      5021.02
6    41.0965811  -81.4897971      5022.00
7    41.0829912  -81.4760965      5023.00
8    41.0697756  -81.4811058      5025.00
9    41.0687054   -81.469198      5026.00
10   41.0788732  -81.4500049      5027.00
11    41.068563  -81.4470118      5028.00
12    41.068852  -81.4967052      5031.00
13   41.0601438  -81.4955638      5032.00
14    41.040496  -81.4971586      5033.00
15    41.063034   -81.486175      5034.00
16    41.049353  -81.4838909      5035.00
17   41.0533235  -81.4659917      5036.00
18   41.0522139  -81.4461962      5037.01
19   41.0526704  -81.4303702      5037.02
20   41.0291706  -81.4727794      5038.00
21    41.065477  -81.5056081      5041.00
22   41.0606598  -81.5139659      5042.00
23   41.0499564  -81.5233459      5044.00
24   41.0510032  -81.5142789      5045.00
25   41.0536023  -81.5050114      5046.00
26   41.0432782  -81.5135793      5047.00
27   41.0343632   -81.516324      5048.00
28   41.0615779  -81.5458687      5052.00
29    41.058939  -81.5354627      5053.00
..          ...          ...          ...
105  40.9457792  -81.5817439      5317.02
106  41.0184964   -81.537607      5318.01
107  41.0094637  -81.5074864      5318.02
108  41.0086144   -81.649467      5320.01
109  41.0451103  -81.6648842      5320.03
110  41.0440182  -81.6127542      5320.04
111  41.1243788  -81.6176923      5322.02
112   41.177394  -81.6627095      5323.01
113  41.1616527  -81.6109586      5323.02
114  41.2577903  -81.6401234      5325.01
115  41.2228382   -81.631974      5325.02
116  41.2417357   -81.540333      5326.00
117  41.3008659  -81.4858535      5327.01
118  41.3388568  -81.5268406      5327.02
119  41.2874758  -81.5325423      5327.03
120  41.3340145  -81.5655108      5327.05
121  41.3008741  -81.5650318      5327.06
122  41.3269173  -81.4922085      5327.08
123  41.1828813  -81.5530077      5329.01
124  41.1771447  -81.5109597      5329.02
125  40.9547135  -81.4360888      5329.99
126  41.0164309  -81.5664915      5330.00
127  41.2657958    -81.42083      5331.01
128   41.244439  -81.4167448      5331.02
129  41.2183195   -81.416403      5332.00
130  41.0808405  -81.6157451      5334.00
131  41.1250983  -81.6621756      5335.01
132  41.0904593  -81.6631372      5335.02
133  41.2268768  -81.4707843      5340.00
134  41.2662129   -81.464497      5341.00

[135 rows x 3 columns]

### I then removed all rows which did not correspond to a census tract in Akron.

In [21]:
census_tracts_in_akron = census_tracts[census_tracts['fips number'].isin(census_tracts_in_Akron)]
census_tracts_in_akron

latitude    longitude  fips number
0    41.0903877  -81.5058552      5011.00
1    41.0683703  -81.5140053      5017.00
2    41.0688191  -81.5429231      5018.00
3    41.0656521  -81.5326908      5019.00
4    41.1091563   -81.499983      5021.01
5    41.1097084   -81.478352      5021.02
6    41.0965811  -81.4897971      5022.00
7    41.0829912  -81.4760965      5023.00
8    41.0697756  -81.4811058      5025.00
9    41.0687054   -81.469198      5026.00
10   41.0788732  -81.4500049      5027.00
11    41.068563  -81.4470118      5028.00
12    41.068852  -81.4967052      5031.00
13   41.0601438  -81.4955638      5032.00
14    41.040496  -81.4971586      5033.00
15    41.063034   -81.486175      5034.00
16    41.049353  -81.4838909      5035.00
17   41.0533235  -81.4659917      5036.00
18   41.0522139  -81.4461962      5037.01
19   41.0526704  -81.4303702      5037.02
20   41.0291706  -81.4727794      5038.00
21    41.065477  -81.5056081      5041.00
22   41.0606598  -81.5139659      5042.00
23   41.0499564  -81.5233459      5044.00
24   41.0510032  -81.5142789      5045.00
25   41.0536023  -81.5050114      5046.00
26   41.0432782  -81.5135793      5047.00
27   41.0343632   -81.516324      5048.00
28   41.0615779  -81.5458687      5052.00
29    41.058939  -81.5354627      5053.00
..          ...          ...          ...
41    41.072669   -81.549932      5067.00
42   41.0607285  -81.5227705      5068.00
43   41.1253213  -81.5768584      5071.01
44   41.1145308  -81.5968219      5071.02
45   41.1232088  -81.5592463      5072.01
46   41.1098284  -81.5578894      5072.02
47   41.1205242  -81.5419881      5072.03
48   41.0993344  -81.5386249      5073.00
49    41.096062  -81.5246588      5074.00
50    41.121189  -81.5241664      5075.00
51   41.1179371  -81.5035892      5076.00
52   41.1491105  -81.5566533      5080.00
53   41.0825851  -81.5223112      5083.01
54   41.0545099  -81.5736184      5083.99
55   41.0806405  -81.5509026      5086.00
56   41.0730965  -81.5746248      5088.00
57   41.0772923  -81.5058328      5089.00
58   41.0797942  -81.4938454      5090.00
71   41.1442058  -81.5174243      5203.01
94   41.0539128  -81.4107369      5310.01
95   41.0320304  -81.4159202      5310.02
96   41.0204442  -81.4387054      5311.01
106  41.0184964   -81.537607      5318.01
107  41.0094637  -81.5074864      5318.02
111  41.1243788  -81.6176923      5322.02
113  41.1616527  -81.6109586      5323.02
123  41.1828813  -81.5530077      5329.01
124  41.1771447  -81.5109597      5329.02
126  41.0164309  -81.5664915      5330.00
130  41.0808405  -81.6157451      5334.00

[71 rows x 3 columns]

In [22]:
library_info

latitude  longitude                                               name
0   41.083365 -81.516839    Akron-Summit County Public Library: Main Branch
1   41.134771 -81.485502                             Cuyahoga Falls Library
2   41.115915 -81.574477             Northwest Akron Branch Library (ASCPL)
3   41.054888 -81.440852         Akron Summit County Library - Ellet Branch
4   41.160995 -81.440889                   Stow Munroe Falls Public Library
5   41.136310 -81.621572  Akron-Summit County Public Library - Fairlawn/...
6   41.241590 -81.443457                Hudson Library & Historical Society
7   41.314069 -81.626122  Cuyahoga County Public Library - Brecksville B...
8   41.331228 -81.721279  Cuyahoga County Public Library - North Royalto...
9   41.042591 -81.514704                             Firestone Park Library
10  41.239848 -81.817134  Medina County District Library - Brunswick Branch
11  41.137700 -81.861865     Medina County District Library - Medina Branch
12  40.807501 -81.457350  Perry Sippo Branch - Stark County District Lib...
13  41.383238 -81.732032      Cuyahoga County Public Library - Parma Branch
14  41.404598 -81.697316  Cuyahoga County Public Library - Parma-Snow Br...
15  41.043814 -81.558175        Akron-Summit Public Library: Kenmore Branch
16  41.376848 -81.646840  Cuyahoga County Public Library - Independence ...
17  41.070995 -81.544334  Akron-Summit County Public Library: Odom Boule...
18  41.070995 -81.544334  Akron-Summit County Public Library: Odom Boule...
19  41.108272 -81.509716  Akron-Summit County Public Library–North Hill ...
20  41.096652 -81.543462  Akron-Summit County Public Library: Highland S...
21  41.036160 -81.564450         Akron summit county library Kenmore branch
22  41.066634 -81.481187  Akron-Summit County Public Library: Goodyear B...
23  41.434109 -81.735940   Cuyahoga County Public Library - Brooklyn Branch
24  41.495701 -81.564678  Cleveland Heights-University Heights Public Li...
25  41.440737 -81.526012  Cuyahoga County Public Library-Warrensville He...
26  41.501488 -81.690829             Cleveland Public Library - Main Branch
27  41.465750 -81.566419        Shaker Heights Public Library - Main Branch
28  41.484666 -81.804208              Lakewood Public Library - Main Branch
29  41.555777 -81.440938                           Mayfield Village Library
30  41.413617 -81.925711  Cuyahoga County Public Library - North Olmsted...
31  41.447664 -81.924587                     Westlake Porter Public Library
32  40.797588 -81.942049                        Wayne County Public Library
33  41.443797 -81.855891  Cuyahoga County Public Library - Fairview Park...
34  41.083678 -81.565000           Akron-Summit County Library-Maple Valley
35  41.665376 -81.346765                Mentor Public Library - Main Branch
36  41.031620 -81.639250                              Norton Branch Library
37  41.509895 -82.012773                           Avon Lake Public Library
38  41.639450 -81.407296                          Willoughby Public Library
39  41.727680 -81.243029                                     Morley Library
40  41.109375 -81.568264                                   NW Akron Library

### I then used the latitude and longitude of all libraries and the center of each census tract to determine the distance to the closest library for each census tract.

In [23]:
census_tract_lat = list(census_tracts_in_akron['latitude'])
census_tract_long = list(census_tracts_in_akron['longitude'])
library_lat = list(library_info['latitude'])
library_long = list(library_info['longitude'])


In [24]:
# uses formula described at https://www.geeksforgeeks.org/program-distance-two-points-earth/
from math import radians, cos, sin, asin, sqrt
def determine_distance(lat1,lat2,long1,long2):
    lat1 = radians(float(lat1))
    lat2 = radians(float(lat2))
    long1 = radians(float(long1))
    long2 = radians(float(long2))
    
    dlon = long2 - long1
    dlat = lat2 - lat1
    a = sin(dlat/2)**2 + cos(lat1) * cos(lat2) * sin(dlon/2)**2
    c = 2 * asin(sqrt(a))
    r = 6371
    return(c * r)
def shortest_dist(list_name):
    list_name.sort()
    return(list_name[0])


In [25]:
counter_tract = 0
closest_library_km = []
while counter_tract < len(census_tract_lat):
    counter_library = 0
    library_dis = []
    while counter_library < len(library_lat):
        library_dis.append(determine_distance(census_tract_lat[counter_tract],library_lat[counter_library],census_tract_long[counter_tract],library_long[counter_library]))
        counter_library += 1
    counter_tract += 1
    closest_library_km.append(shortest_dist(library_dis))
print(closest_library_km)
    

[1.2071711586529288, 1.6841667289757698, 0.2693598428570056, 1.1427189704846819, 0.821350579962201, 2.632555560131928, 2.11552610120139, 1.8682550431923755, 0.3494338869811138, 1.0311307962207004, 2.7752480589760182, 1.6059247541131552, 1.3241380113151366, 1.4048618279116214, 1.4898314812250224, 0.5788759974188102, 1.9348513358640989, 1.9528375601718633, 0.5377894130583718, 0.9128358065781642, 3.819971556040501, 2.051409904919515, 2.010066683222186, 1.0935562381686754, 0.9360278905654938, 1.4696030887286122, 0.1213775677479414, 0.924966260846211, 1.0550408541110963, 1.5331087838008077, 0.6353143559669402, 0.8522955479263054, 1.778083778179999, 0.4748680269625791, 0.6395371123780261, 1.9126916314139792, 0.704992786536212, 0.8257529532791393, 0.48927789620528583, 1.052410937494142, 1.0162003194824487, 0.5047945285076149, 2.127155677685541, 1.0647725073463215, 1.8781841467219131, 1.5118230664053156, 0.8706374298019403, 2.5263156636818724, 0.5032438482053316, 1.556517357254025, 1.878412399

### I then added the number of kilometers to the nearest library before sorting my dataframe with the rows with the largest distance to a library at the top before taking the top five furthest tracts from a library and saving them as their own dataframe.

In [26]:
census_tracts_in_akron.insert(3,'km to library', closest_library_km)
library_and_tract = census_tracts_in_akron
library_and_tract

latitude    longitude  fips number  km to library
0    41.0903877  -81.5058552      5011.00       1.207171
1    41.0683703  -81.5140053      5017.00       1.684167
2    41.0688191  -81.5429231      5018.00       0.269360
3    41.0656521  -81.5326908      5019.00       1.142719
4    41.1091563   -81.499983      5021.01       0.821351
5    41.1097084   -81.478352      5021.02       2.632556
6    41.0965811  -81.4897971      5022.00       2.115526
7    41.0829912  -81.4760965      5023.00       1.868255
8    41.0697756  -81.4811058      5025.00       0.349434
9    41.0687054   -81.469198      5026.00       1.031131
10   41.0788732  -81.4500049      5027.00       2.775248
11    41.068563  -81.4470118      5028.00       1.605925
12    41.068852  -81.4967052      5031.00       1.324138
13   41.0601438  -81.4955638      5032.00       1.404862
14    41.040496  -81.4971586      5033.00       1.489831
15    41.063034   -81.486175      5034.00       0.578876
16    41.049353  -81.4838909      5035.00       1.934851
17   41.0533235  -81.4659917      5036.00       1.952838
18   41.0522139  -81.4461962      5037.01       0.537789
19   41.0526704  -81.4303702      5037.02       0.912836
20   41.0291706  -81.4727794      5038.00       3.819972
21    41.065477  -81.5056081      5041.00       2.051410
22   41.0606598  -81.5139659      5042.00       2.010067
23   41.0499564  -81.5233459      5044.00       1.093556
24   41.0510032  -81.5142789      5045.00       0.936028
25   41.0536023  -81.5050114      5046.00       1.469603
26   41.0432782  -81.5135793      5047.00       0.121378
27   41.0343632   -81.516324      5048.00       0.924966
28   41.0615779  -81.5458687      5052.00       1.055041
29    41.058939  -81.5354627      5053.00       1.533109
..          ...          ...          ...            ...
41    41.072669   -81.549932      5067.00       0.504795
42   41.0607285  -81.5227705      5068.00       2.127156
43   41.1253213  -81.5768584      5071.01       1.064773
44   41.1145308  -81.5968219      5071.02       1.878184
45   41.1232088  -81.5592463      5072.01       1.511823
46   41.1098284  -81.5578894      5072.02       0.870637
47   41.1205242  -81.5419881      5072.03       2.526316
48   41.0993344  -81.5386249      5073.00       0.503244
49    41.096062  -81.5246588      5074.00       1.556517
50    41.121189  -81.5241664      5075.00       1.878412
51   41.1179371  -81.5035892      5076.00       1.190989
52   41.1491105  -81.5566533      5080.00       3.981595
53   41.0825851  -81.5223112      5083.01       0.466739
54   41.0545099  -81.5736184      5083.99       1.758286
55   41.0806405  -81.5509026      5086.00       1.205557
56   41.0730965  -81.5746248      5088.00       1.426662
57   41.0772923  -81.5058328      5089.00       1.143263
58   41.0797942  -81.4938454      5090.00       1.807618
71   41.1442058  -81.5174243      5203.01       2.871744
94   41.0539128  -81.4107369      5310.01       2.527488
95   41.0320304  -81.4159202      5310.02       3.291149
96   41.0204442  -81.4387054      5311.01       3.834180
106  41.0184964   -81.537607      5318.01       2.987907
107  41.0094637  -81.5074864      5318.02       3.733064
111  41.1243788  -81.6176923      5322.02       1.365915
113  41.1616527  -81.6109586      5323.02       2.954754
123  41.1828813  -81.5530077      5329.01       7.660222
124  41.1771447  -81.5109597      5329.02       5.171389
126  41.0164309  -81.5664915      5330.00       2.200401
130  41.0808405  -81.6157451      5334.00       4.264885

[71 rows x 4 columns]

In [27]:
library_and_tract_sorted = library_and_tract.sort_values(by='km to library', ascending=False)
library_and_tract_sorted

latitude    longitude  fips number  km to library
123  41.1828813  -81.5530077      5329.01       7.660222
124  41.1771447  -81.5109597      5329.02       5.171389
130  41.0808405  -81.6157451      5334.00       4.264885
52   41.1491105  -81.5566533      5080.00       3.981595
96   41.0204442  -81.4387054      5311.01       3.834180
20   41.0291706  -81.4727794      5038.00       3.819972
107  41.0094637  -81.5074864      5318.02       3.733064
95   41.0320304  -81.4159202      5310.02       3.291149
106  41.0184964   -81.537607      5318.01       2.987907
113  41.1616527  -81.6109586      5323.02       2.954754
71   41.1442058  -81.5174243      5203.01       2.871744
10   41.0788732  -81.4500049      5027.00       2.775248
5    41.1097084   -81.478352      5021.02       2.632556
94   41.0539128  -81.4107369      5310.01       2.527488
47   41.1205242  -81.5419881      5072.03       2.526316
126  41.0164309  -81.5664915      5330.00       2.200401
42   41.0607285  -81.5227705      5068.00       2.127156
6    41.0965811  -81.4897971      5022.00       2.115526
21    41.065477  -81.5056081      5041.00       2.051410
22   41.0606598  -81.5139659      5042.00       2.010067
17   41.0533235  -81.4659917      5036.00       1.952838
16    41.049353  -81.4838909      5035.00       1.934851
35   41.0379172  -81.5366519      5059.00       1.912692
50    41.121189  -81.5241664      5075.00       1.878412
44   41.1145308  -81.5968219      5071.02       1.878184
7    41.0829912  -81.4760965      5023.00       1.868255
58   41.0797942  -81.4938454      5090.00       1.807618
32   41.0521181  -81.5400555      5056.00       1.778084
54   41.0545099  -81.5736184      5083.99       1.758286
1    41.0683703  -81.5140053      5017.00       1.684167
..          ...          ...          ...            ...
55   41.0806405  -81.5509026      5086.00       1.205557
51   41.1179371  -81.5035892      5076.00       1.190989
57   41.0772923  -81.5058328      5089.00       1.143263
3    41.0656521  -81.5326908      5019.00       1.142719
23   41.0499564  -81.5233459      5044.00       1.093556
43   41.1253213  -81.5768584      5071.01       1.064773
28   41.0615779  -81.5458687      5052.00       1.055041
39   41.0879242  -81.5483198      5065.00       1.052411
9    41.0687054   -81.469198      5026.00       1.031131
40   41.0895499  -81.5358304      5066.00       1.016200
24   41.0510032  -81.5142789      5045.00       0.936028
27   41.0343632   -81.516324      5048.00       0.924966
19   41.0526704  -81.4303702      5037.02       0.912836
46   41.1098284  -81.5578894      5072.02       0.870637
31   41.0503538   -81.552875      5055.00       0.852296
37   41.0878974  -81.5731083      5062.00       0.825753
4    41.1091563   -81.499983      5021.01       0.821351
36    41.105615   -81.575039      5061.00       0.704993
34   41.0310989  -81.5608263      5058.00       0.639537
30   41.0479107   -81.563456      5054.00       0.635314
15    41.063034   -81.486175      5034.00       0.578876
18   41.0522139  -81.4461962      5037.01       0.537789
41    41.072669   -81.549932      5067.00       0.504795
48   41.0993344  -81.5386249      5073.00       0.503244
38   41.0952284  -81.5489868      5064.00       0.489278
33   41.0403027  -81.5613986      5057.00       0.474868
53   41.0825851  -81.5223112      5083.01       0.466739
8    41.0697756  -81.4811058      5025.00       0.349434
2    41.0688191  -81.5429231      5018.00       0.269360
26   41.0432782  -81.5135793      5047.00       0.121378

[71 rows x 4 columns]

In [28]:
five_furthest_from_library = library_and_tract_sorted.head()
five_furthest_from_library

latitude    longitude  fips number  km to library
123  41.1828813  -81.5530077      5329.01       7.660222
124  41.1771447  -81.5109597      5329.02       5.171389
130  41.0808405  -81.6157451      5334.00       4.264885
52   41.1491105  -81.5566533      5080.00       3.981595
96   41.0204442  -81.4387054      5311.01       3.834180

In [29]:
five_fips = list(five_furthest_from_library['fips number'])
five_tracts = []
for item in five_fips:
    five_tracts.append(int(item * 100))
five_tracts

[532901, 532902, 533400, 508000, 531101]

### I then used the U.S. Census API with data from the 2018  5- year American Community Survey to determine the rate of highschool graduation by the age of 25, what percent of the population had broadband internet access at home, and the percent of people below the poverty line in each of the five remaining tracts.

In [30]:
! pip install requests

In [31]:
import requests

In [32]:
#Highschool graduation rate by 25
base_url_hs_first = 'https://api.census.gov/data/2018/acs/acs5/profile?get=DP02_0066PE&for=tract:'
base_url_hs_sec = '&in=state:39%20county:153'
#&key=YOUR_KEY_GOES_HERE

# percent with internet connection
base_url_internet_first = 'https://api.census.gov/data/2018/acs/acs5/profile?get=DP02_0152PE&for=tract:'
base_url_internet_sec = '&in=state:39%20county:153'
#&key=YOUR_KEY_GOES_HERE

#percent below poverty level
base_url_poverty_first = 'https://api.census.gov/data/2018/acs/acs5/profile?get=DP03_0128PE&for=tract:'
base_url_poverty_sec = '&in=state:39%20county:153'
#&key=YOUR_KEY_GOES_HERE



In [33]:
hs_graduation_rates = []
internet_connection = []
poverty_level = []

for tract in five_tracts:
    
    url_hs = base_url_hs_first + str(tract) + base_url_hs_sec
    url_internet = base_url_internet_first + str(tract) + base_url_internet_sec
    url_poverty = base_url_poverty_first + str(tract) + base_url_poverty_sec
    
    hs = requests.get(url_hs)
    internet = requests.get(url_internet)
    poverty = requests.get(url_poverty)
    
    hs_json = hs.json()
    internet_json = internet.json()
    poverty_json = poverty.json()
    
    hs_graduation_rates.append(hs_json[1][0])
    internet_connection.append(internet_json[1][0])
    poverty_level.append(poverty_json[1][0])


### I then used this data to create a dataframe for all the data mentioned above in addition to the distance to the nearest libraries in Km and then I used this data to determine which Census Tract the library should be in.

In [34]:
census_grad_df = pd.DataFrame(list(zip(five_tracts, hs_graduation_rates)), columns = ['Census Tract', '% Graduated from Highschool by 25'])
census_grad_df


Census Tract % Graduated from Highschool by 25
0        532901                              98.1
1        532902                              94.2
2        533400                              94.5
3        508000                              99.1
4        531101                              87.1

In [35]:
census_internet_df = pd.DataFrame(list(zip(five_tracts, internet_connection)), columns = ['Census Tract', '% With Broadband Internet'])
census_internet_df


Census Tract % With Broadband Internet
0        532901                      83.1
1        532902                      83.1
2        533400                      75.1
3        508000                      89.1
4        531101                      79.0

In [36]:
census_poverty_df = pd.DataFrame(list(zip(five_tracts, poverty_level)), columns = ['Census Tract', '% In Poverty'])
census_poverty_df

Census Tract % In Poverty
0        532901          6.9
1        532902          8.6
2        533400          8.5
3        508000          9.9
4        531101         11.3

In [37]:
library_km = list(five_furthest_from_library['km to library'])
distance_df = pd.DataFrame(list(zip(five_tracts, library_km)), columns = ['Census Tract', 'Distance to Library in Kilometers'])
distance_df

Census Tract  Distance to Library in Kilometers
0        532901                           7.660222
1        532902                           5.171389
2        533400                           4.264885
3        508000                           3.981595
4        531101                           3.834180

### Decision for Library Placement and Justification

Altogether, while tract 531101 has the shortest distance to a library, its residents are more likely to live below the poverty line and to not have graduated highschool by 25 than people in other tracts, and they also have the second-lowest rate of broadband internet access. Since the census tracts do not line up with city boundaries, the center of this tract is outside of Akron, but the parts of this tract which are inside the city would benefit from the services a library could provide.